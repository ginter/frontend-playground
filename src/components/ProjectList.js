import React from 'react'

const ProjectList = ({ projects, deleteProject }) => {
  if (projects.length === 0) return <p>No projects found</p>;

  return(
    <ul>
      {projects.map(project => (
        <li key={project.uid} className='project-row'>
          {project.name}
          <button onClick={() => deleteProject(project.uid)}>DELETE</button>
        </li>
        )
      )}
    </ul>
  );
}

export default ProjectList
