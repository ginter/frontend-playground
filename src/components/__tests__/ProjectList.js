import React from 'react'
import renderer from 'react-test-renderer'

import ProjectList from '../ProjectList'

jest.unmock('../ProjectList');

describe('ProjectList', () => {
  it('renders correctly with no projects', () => {
    const component = renderer.create(<ProjectList projects={[]} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a single project', () => {
    const project = { uid: 'projId', name: 'projName' };
    const component = renderer.create(<ProjectList projects={[project]} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
