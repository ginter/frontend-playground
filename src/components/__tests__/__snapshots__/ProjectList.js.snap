exports[`ProjectList renders correctly with a single project 1`] = `
<ul>
  <li
    className="project-row">
    <p>
      projName
    </p>
    <button
      onClick={[Function onClick]}>
      DELETE
    </button>
  </li>
</ul>
`;

exports[`ProjectList renders correctly with no projects 1`] = `
<p>
  No projects found
</p>
`;
